### DISCUSSION TITLE
Convert dot format with subgraphs into json

### DISCUSSION THREAD

**Discussion by divyansshhh** [networkx/networkx#5254] [Q&A] [answered]
Hi, I was trying to convert graphs in dot format to json and I'm facing some issues with that. Consider the following dot graph - 

```
digraph G {

	subgraph cluster_0 {
		style=filled;
		color=lightgrey;
		node [style=filled,color=white];
		a0 -> a1 -> a2 -> a3;
		label = "process #1";
	}

	subgraph cluster_1 {
		node [style=filled];
		b0 -> b1 -> b2 -> b3;
		label = "process #2";
		color=blue
	}
	start -> a0;
	start -> b0;
	a1 -> b3;
	b2 -> a3;
	a3 -> a0;
	a3 -> end;
	b3 -> end;

	start [shape=Mdiamond];
	end [shape=Msquare];
}
```

If I use `networkx` to convert it into json using  - 
```
import networkx as nx
from networkx.readwrite import json_graph
dot_graph = nx.nx_pydot.read_dot(sys.argv[1])
print(json.dumps(json_graph.node_link_data(dot_graph)))
```
and found that the resulting json does not contain any keys/nodes related to the subgraphs. Is there a way to get subgraph information?

**Comment by MridulS** [comment, marked as answer]
There isn't really an equivalent of a `dot` subgraph, atleast in the current implementation. This can be implemented but currently we only parse the nodes and edges objects from the `dot` format. We use `pydot` (https://github.com/pydot/pydot) underneath which does parse the information about dot `subgraph`, so one way to achieve this to write your own convertor from `pydot.SubGraph` to NetworkX Graph and then use the JSON convertor.

And well if you end up creating a convertor for that, maybe we could also add it to NetworkX!
